contract.md

```diff
@@ -12,29 +17,60 @@ like to work on, as part of your project (for example, other things related to D
 you have come across, or topics from other classes)? Why are you interested in working on
 these things?_
 
+I think one of the things I’m most interested in is transferring the knowledge I have in Scala and DSL design to other host languages. I think that if I can get an idea of how to write, for instance, parser combinators in other languages, or more broadly, how to write external DSL’s in other languages, I should be able to transfer that knowledge to other languages in the future and could almost lessen the tradeoffs when deciding on a host language in the future. 
+	I also want to further grow my experience in Scala. While I do feel comfortable working in Scala, I do think my Piconot assignment was compromised and sacrificed much of what I wanted in it due to not knowing the language or libraries well enough. I think that working on those skills will be beneficial to this project and to other projects in the future. 
+	I mentioned earlier that I’d want to work on a frontend user interface for this project to preview changes, but it is completely unnecessary and I would only work on it should time allow for it. As for concepts and skills not covered in class, I think investigating other host languages would be useful. While I would like to work on this project using Scala, if I investigate other host languages and find others that are more suitable for this project I would like to use those instead, which I don’t think has been covered in class. 
+	I’m most interested in working on these things because I either want to grow my experience in DSL design and Scala, as well as learn new ideas and techniques that will be useful in future projects. Transferring skills to other host languages, learning more about Scala, learning more about external DSL design, growing my experience in UX design are all things that are important to me and would definitely be beneficial going forward.
+
```

Comment:
As an offering for host languages and frontend - if it's ok that the frontend is not ridiculously professional looking, then PyQt in Python is a free and simple way to make a GUI, and could be a option if you get to that point.